lib/ood_core/job/adapters/slurm.rb

```diff
@@ -178,6 +179,18 @@ def get_jobs(id: "", owner: nil, attrs: nil)
             return [{ id: id, state: 'undetermined' }]
           end
 
+          def accounts
+            user = Etc.getlogin
+            args = ['-np', 'show', 'accounts', 'withassoc', 'format=account,user', '-P']
```

Comment:
This will take a VERY long time on OSC as we have thousands of accounts and even more users.  You will almost certainly want the sacctmgr command to have `where user=<username>` as an argument otherwise you dump all users which at OSC could take a minute or two or even longer.

Reply:
Also why use `-p` and `-P` in same command, it's either one of the other , not both.  Using just `-p` has trailing pipe where `-P` does not.

Ran this command at OSC and faster than expected but still took 5 seconds.  But using my username of `where user=tdockendorf` took 0.5 seconds.

Reply:
Yea thanks for those both, especially the `where` clause as I was seeing it take some time (was going to cache it in the OOD layer).